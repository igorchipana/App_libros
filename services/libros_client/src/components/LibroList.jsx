import React from 'react';
 

const styles = {
    color: "black",
    marginTop: "50px",
    marginLeft: "0px",
    float: "center",
    fontSize: "18px"
};
const LibroList = (props) => {
  return (

    <div  className="well" style={styles}> 
      <legend className="text-center header" >Lista de Libros registrados</legend>
              <div>
                      <table className="table table-striped"  >
                      <thead className="thead-dark">
                            <tr>
                              <th scope="col">Nombre del Libro</th>
                              <th scope="col">Descripci√≥n</th>
                              <th scope="col">Categoria</th>
                              <th scope="col">Autor</th>
                              <th scope="col">Precio</th>
                            </tr>
                          </thead>
     
      {
                
        props.libros.map((libro) => {
          return (
                        <tbody >
                      <tr>
                       <td key={libro.id}>{libro.nombre}</td>
                       <td>{libro.descripcion}</td>
                       <td>{libro.categoria}</td>
                       <td>{libro.autor}</td>
                       <td>{libro.costo}</td>
                       </tr>
                        </tbody> 
          )
        })

      }
                </table>
                     
        </div>

    </div>

 
  )
};
 
export default LibroList;

