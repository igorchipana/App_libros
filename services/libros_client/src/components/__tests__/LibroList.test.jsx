import React from 'react';
import { shallow } from 'enzyme';
 
import LibroList from '../LibroList';
 
const libros = [
  {
	'active': true,
	'id':1,
	'nombre': 'los amantes son dementes',
	'costo': '20',
	'autor': 'mauricio',
	'descripcion': 'buen libro',
	'categoria': 'drama'
  }
];


test('LibroList renders properly', () => {
  const wrapper = shallow(<LibroList libros={libros}/>);
  const element = wrapper.find('td');
  expect(element.length).toBe(5);
  expect(element.get(0).props.children).toBe('los amantes son dementes');
  expect(element.get(1).props.children).toBe('buen libro');
  expect(element.get(2).props.children).toBe('drama');
  expect(element.get(3).props.children).toBe('mauricio');
  expect(element.get(4).props.children).toBe('20');
});
