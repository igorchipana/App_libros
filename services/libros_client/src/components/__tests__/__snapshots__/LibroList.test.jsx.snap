// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LibroList renders a snapshot properly 1`] = `
<div
  className="well"
  style={
    Object {
      "color": "black",
      "float": "center",
      "fontSize": "18px",
      "marginLeft": "0px",
      "marginTop": "50px",
    }
  }
>
  <legend
    className="text-center header"
  >
    Lista de Libros registrados
  </legend>
  <div>
    <table
      className="table table-striped"
    >
      <thead
        className="thead-dark"
      >
        <tr>
          <th
            scope="col"
          >
            Nombre del Libro
          </th>
          <th
            scope="col"
          >
            Descripci√≥n
          </th>
          <th
            scope="col"
          >
            Categoria
          </th>
          <th
            scope="col"
          >
            Autor
          </th>
          <th
            scope="col"
          >
            Precio
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            los amantes son dementes
          </td>
          <td />
          <td />
          <td />
          <td>
            20
          </td>
        </tr>
      </tbody>
    </table>
  </div>
</div>
`;
